File: app/src/main/java/fragment/FragAzul.java
package fragment;

import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.example.exemplo.R;

public class FragAzul extends Fragment {

    private EditText etAzul;
    private Button buttonAzul;
    private TextView textAzul;

    ActivityContract activity;

    public static FragAzul newInstance(Bundle bundle) {
        FragAzul frag = new FragAzul();
        frag.setArguments(bundle);
        return frag;
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.frag_azul, container, false);
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        if (context instanceof ActivityContract) activity = (ActivityContract) context;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        etAzul = view.findViewById(R.id.et_azul);
        buttonAzul = view.findViewById(R.id.button_azul);
        textAzul = view.findViewById(R.id.texto_azul);

        buttonAzul.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                activity.setTextVermelho(etAzul.getText().toString());
            }
        });
    }

    public void setTextAzul(String texto) {
        textAzul.setText(texto);
    }
}

File: app/src/main/java/fragment/FragAzul.kt
package fragment

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import androidx.fragment.app.Fragment
import com.example.exemplo.R

class FragAzul : Fragment() {
    private lateinit var etAzul: EditText
    private lateinit var buttonAzul: Button
    private lateinit var textAzul: TextView
    private lateinit var activity: ActivityContract

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View = inflater.inflate(R.layout.frag_azul, container, false)

    override fun onAttach(context: Context) {
        super.onAttach(context)
        if (context is ActivityContract) activity =
            context
    }

    override fun onViewCreated(
        view: View,
        savedInstanceState: Bundle?
    ) {
        super.onViewCreated(view, savedInstanceState)
        etAzul = view.findViewById(R.id.et_azul)
        buttonAzul = view.findViewById(R.id.button_azul)
        textAzul = view.findViewById(R.id.texto_azul)
        buttonAzul.setOnClickListener {
            activity.setTextVermelho(etAzul.text.toString())
        }
    }

    fun setTextAzul(texto: String) {
        textAzul.text = texto
    }

    companion object {
        @JvmStatic fun newInstance(bundle: Bundle? = null): FragAzul {
            val frag = FragAzul()
            frag.arguments = bundle
            return frag
        }
    }
}